apps/librelingo_json_export/librelingo_json_export/challenge_types.py

```diff
@@ -92,6 +92,27 @@ def get_options_challenge(phrase, _):
         }
     ]
 
+def get_pairing_challenge(words, _):
+    if not isinstance(words, list):
+        words = [words]
+
+    return [
+        {
+            "type": "pairing",
+            "pairs" : [
+                {
+                    "formInTargetLanguage": remove_control_characters_for_display(
+                        word.in_target_language[0]),
+                    "meaningInSourceLanguage": remove_control_characters_for_display(
+                        word.in_source_language[0])
+                }
+            for word in words],
```

Comment:
one more thing that's missing here is filtering out invalid pairs (such as when one thing has a translation that is also the translation of something else, and similar cases).

This PR can be merged even without that, because it won't be included in the frontend yet anyway. But it would be nice if you implemented this in either this PR or a second PR :pray: